anta/tests/configuration.py

```diff
@@ -75,3 +76,55 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(command_output)
+
+
+class VerifyRunningConfigLines(AntaTest):
+    """Verifies the given regular expression patterns are present in the running-config.
+
+    **Since this uses regular expression searches on the whole running-config, it can
+    drastically impact performance and should only be used if no other tests is available for.**
+    If possible, try using another Anta test that is more specific.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all the patterns are found in the running-config.
+    * Failure: The test will fail if any of the patterns are NOT found in the running-config.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.configuration:
+      - VerifyRunningConfigLines:
+            regex_patterns:
+                - "^enable password.*$"
+                - "bla bla"
+    ```
+    """
+
+    name = "VerifyRunningConfigLines"
+    description = "Search the Running-Config for the given RegEx patterns."
+    categories: ClassVar[list[str]] = ["configuration"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show running-config", ofmt="text")]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyRunningConfigLines test."""
+
+        regex_patterns: list[str]
+        """List of regular expressions."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyRunningConfigLines."""
+        failure_msgs = []
+        command_output = self.instance_commands[0].text_output
+
+        for pattern in self.inputs.regex_patterns:
+            re_search = re.compile(pattern, flags=re.MULTILINE)
```

Comment:
This line here will generate an error if the regex string is invalid (e.g. `'['`). We can catch the error when ANTA parses the inputs with a custom type.

In the `anta.custom_types` module, add something like this:
```
def validate_regex(value: str) -> str:
    """Validate that the input value is a valid regex format."""
    try:
        re.compile(value)
    except re.error as e:
        msg = f"Invalid regex: {e}"
        raise ValueError(msg) from e
    return value
 ```
 And then create the custom type in the same file: `RegexString = Annotated[str, AfterValidator(validate_regex)]`
 
 You can now use that custom type on line 112 above.
 
With that in place we can catch a bad regex from the start.

Reply:
This is great. I did try to find a way to only compile the regex once, however that wouldn't be a validator. At the end I went your suggestion.